client/lib/currency-formatter/index.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import formatter from 'currency-formatter';
+import get from 'lodash/get';
+
+// override symbol defaults for currency code
+const symbolMap = {
+	CAD: 'C$',
+	AUD: 'A$'
+};
+
+const thousandsMap = {
+	EUR: '.'
+};
+
+/**
+ * Formats money with a given currency code
+ * @param   {Number}     number              number to format
+ * @param   {Object}     options             options object
+ * @param   {String}     options.code        currency code e.g. 'USD'
+ * @param   {String}     options.symbol      currency symbol e.g. 'A$'
+ * @param   {String}     options.decimal     decimal symbol e.g. ','
+ * @param   {String}     options.thousand    thousands separator
+ * @param   {Number}     options.precision   decimal digits
+ * @param   {String}     options.format      a custom format string
+ * @returns {String}                         A formatted string.
+ */
+export default function currencyFormatter( number, options = {} ) {
+	const { symbol, thousandsSeparator } = findCurrency( options.code );
+	return formatter.format( number, Object.assign( { symbol, thousand: thousandsSeparator }, options ) );
```

Comment:
Please don't feel like I'm trying to be critical - this is just a style note related to L41 below, but you can avoid the `Object.assign()` here too. This is just my personal preference and nothing more :smile:

```js
return formatter.format( number, { symbol, thousand: thousandsSeparator, ...options } );
```

If you put `...options` first, then if it has `symbol` or `thousand` properties they will be overwritten.